compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.td

```diff
@@ -45,6 +45,43 @@ def EncodingOpTypeAttr:
 // iree_encoding.packed_storage
 //===---------------------------------------------------------------------===//
 
+def LayoutAttr :
+    IREEEncoding_Attr<"Layout", [
+      DeclareAttrInterfaceMethods<IREEEncoding_SerializableEncodingAttrInterface, [
+        "isSerialized",
+        "isIdentityLayout",
+        "calculateStorageSizeInBytes",
+      ]>
+    ]> {
+  let mnemonic = "layout";
+  let summary = [{Indicates potential serialized layouts}];
+  let description = [{
+    This attribute describes the potential layouts of the encoding. It is an
+    array because a device could have multiple target device.
+
+    It has a verifier that will ensure that all the attributes in layouts
+    implement SerializableEncodingAttrInterface.
+  }];
+
+  let parameters = (ins
+    AttrParameter<"ArrayAttr", "Serialized layouts">:$layouts
+  );
+
+  // Implement custom builders to bypass the verification assertion in Base::get
+  // method, i.e., `assert(succeeded(ConcreteT::verifyInvariants(...)))`.
+  let skipDefaultBuilders = 1;
+  let builders = [
+    AttrBuilder<(ins "ArrayAttr":$layouts)>,
+  ];
+
+  let hasCustomAssemblyFormat = 1;
```

Comment:
Why does this have a custom assembly format?

Reply:
I want to remove `layouts = ` from the IR. I should just do `let assemblyFormat = "`<` $layouts `>`";`. 🤦